src/components/Calculator.js

```diff
@@ -1,57 +1,49 @@
-import React from 'react';
+import React, { useState } from 'react';
 import calculate from '../logic/calculate';
 
-export default class Calculator extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      total: 0,
-      next: null,
-      operation: null,
-    };
-    this.handler = this.handler.bind(this);
-  }
+export default function Calculator() {
+  const [state, setState] = useState({
+    total: 0,
+    next: null,
+    operation: null,
+  });
+  const handler = (e) => {
+    setState((state) => calculate(state, e.target.textContent));
```

Comment:
- [OPTIONAL] It is always a good practice to name your functions and variables with meaningful names to help other people easily understand your code without more difficulties. I would recommend naming this function something like `handleClick`